src/main/java/de/tum/in/www1/artemis/web/rest/CompetencyResource.java

```diff
@@ -276,6 +270,57 @@ public ResponseEntity<Competency> importCompetency(@PathVariable long courseId,
         return ResponseEntity.created(new URI("/api/courses/" + courseId + "/competencies/" + competencyToImport.getId())).body(competencyToImport);
     }
 
+    /**
+     * Imports all competencies of the source course (and optionally their relations) into the another
+     *
+     * @param courseId        the id of the course to import into
+     * @param sourceCourseId  the id of the course to import from
+     * @param importRelations if relations should be imported aswell
+     * @return the ResponseEntity with status 201 (Created) and with body containing the imported competencies
+     * @throws URISyntaxException if the Location URI syntax is incorrect
+     */
+    @PostMapping("/courses/{courseId}/competencies/import-all/{sourceCourseId}")
+    @EnforceAtLeastInstructor
+    public ResponseEntity<List<Competency>> importAllCompetenies(@PathVariable long courseId, @PathVariable long sourceCourseId,
+            @RequestParam(defaultValue = "false") boolean importRelations) throws URISyntaxException {
+        log.info("REST request to all competencies from course {} into course {}", sourceCourseId, courseId);
+
+        if (courseId == sourceCourseId) {
+            throw new ConflictException("Cannot import from a course into itself", "Course", "courseCycle");
+        }
+        var course = courseRepository.findWithEagerCompetenciesByIdElseThrow(courseId);
+        var sourceCourse = courseRepository.findWithEagerCompetenciesByIdElseThrow(sourceCourseId);
+        authorizationCheckService.checkHasAtLeastRoleInCourseElseThrow(Role.INSTRUCTOR, course, null);
+        authorizationCheckService.checkHasAtLeastRoleInCourseElseThrow(Role.EDITOR, sourceCourse, null);
```

Comment:
You use Editor here, but the database request only finds curses where the user is Instructor. Which role is correct?

Reply:
I don't quite get what you mean 🤔 `findWithEagerCompetenciesByIdElseThrow` seems to just get the course (and competencies) irrespective of the users Role?

The endpoint is Annotated with `@EnforceAtLeastInstructor` because the user has to be an instructor for this course, but not the course he is importing from.

Reply:
Also for the above suggestion: I changed the endpoint name to `findByTitleInCoursesWhereInstructorOrEditor` (as that is more fitting). I somehow can't comment on that suggestion 🤷